File: src/main/java/net/google/safebrowsing2/model/Status.java
package net.google.safebrowsing2.model;

import java.util.Date;

public class Status {

	private int time;
	private int wait;
	private int errors;

	public Status(int time, int wait, int errors) {
		super();
		this.time = time;
		this.wait = wait;
		this.errors = errors;
	}

	public int getTime() {
		return time;
	}

	public void setTime(int time) {
		this.time = time;
	}

	public int getWait() {
		return wait;
	}

	public void setWait(int wait) {
		this.wait = wait;
	}

	public int getErrors() {
		return errors;
	}

	public void setErrors(int errors) {
		this.errors = errors;
	}

}

File: src/main/scala/net/google/safebrowsing2/model/Status.scala
package net.google.safebrowsing2.model
import java.util.Calendar
import java.util.Date

class Status(val updateTime: Int, val waitSecs: Int, val errors: Int) {
  
  lazy val waitMs = waitSecs * 1000
  
  lazy val waitDate = new Date(updateTime + waitMs)
  
  lazy val updateDate = new Date(updateTime) 

}